### ISSUE TITLE
Runtime only JAR

### ISSUE THREAD

**Issue by yloiseau** [eclipse/golo-lang#447]
I don't know how easy it would be, but creating a JAR with only the runtime dependencies would allows to create a more lightweight “uberjar/fatjar” of a golo application. AFAICT, the only required external dependency is json-simple. All `org.eclipse.golo` but `runtime` could probably also be removed.

**Comment by yloiseau** [comment]
I'm sure IoT folks like @k33g would be pleased :smile:

**Comment by yloiseau** [comment]
Maybe two artefacts: the current full version `golo-VERSION.jar`, and `golo-runtime-VERSION.jar` containing only the classes needed at runtime (AFAICT `org.eclipse.golo.runtime.*` and `gololang.*`).
I suppose some tweaks  to the `build.gradle` would be enough, but I lacks expertise here; maybe @jponge can confirm?

**Comment by yloiseau** [comment]
linked to #400 (cc @Artpej)

**Comment by Artpej** [comment]
Quickly, the embedded template engine requires the compile part... Maybe there are other things we can't do with the runtime only dépendance. Wdyt?

**Comment by yloiseau** [comment]
Arrgg, yes, the `EvaluationEnvironment` depends on the `GoloClassLoader`...
Ok, so `org.eclipse.golo.runtime.*`, org.eclipse.golo.compiler.*` and `gololang.*` then :smile:
In any way, we don't need the `org.eclipse.golo.cli.*`, and thus no `jcommander` nor `txtmark` (required for golodoc)

**Comment by jponge** [comment]
Well in "IoT" settings where you have the freedom to run a JVM for Golo the Jar size that we have isn't much of an issue.

Not all classes are being loaded from a Jar as classloading is lazy.

If we really want to look into this then perhaps we could start with a complete package dependency analysis?

**Comment by yloiseau** [comment]
Indeed... I opened this issue while trying to create a naive gradle task to create  a uberjar:
```gradle
task uberjar(type: Jar, dependsOn: [':compileJava',':compileGolo']) {
    manifest {
        attributes 'Main-Class': golo.mainModule
    }
    baseName = project.name + '-standalone'
    from files(sourceSets.main.output.classesDir)
    from configurations.runtime.asFileTree.files.collect { zipTree(it) }
}
```

It works, but also includes in the jar txtmark, jcommander, asm and junit! Admittedly, I managed to filter them, but I thought it would be better to have a golo distribution not depending on them, which would allows to create a dedicated task in the gradle plugin.
